hub/core/transform/test_transform.py

```diff
@@ -9,85 +13,73 @@ def fn1(i, mul=1, copy=1):
     d = {}
     d["image"] = np.ones((337, 200)) * i * mul
     d["label"] = np.ones((1,)) * i * mul
-    return [d for _ in range(copy)]
+    return d if copy == 1 else [d for _ in range(copy)]
 
 
 def fn2(sample, mul=1, copy=1):
-    d = {}
-    d["image"] = sample["image"] * mul
-    d["label"] = sample["label"] * mul
-    return [d for _ in range(copy)]
+    d = {"image": sample["image"] * mul, "label": sample["label"] * mul}
+    return d if copy == 1 else [d for _ in range(copy)]
 
 
 def fn3(i, mul=1, copy=1):
     d = {}
     d["image"] = np.ones((1310, 2087)) * i * mul
     d["label"] = np.ones((13,)) * i * mul
-    return [d for _ in range(copy)]
+    return d if copy == 1 else [d for _ in range(copy)]
 
 
 @parametrize_all_dataset_storages
 def test_single_transform_hub_dataset(ds):
-    with CliRunner().isolated_filesystem():
-        with Dataset("./test/transform_hub_in") as data_in:
-            data_in.create_tensor("image")
-            data_in.create_tensor("label")
-            for i in range(100):
-                data_in.image.append(i * np.ones((100, 100)))
-                data_in.label.append(i * np.ones((1,)))
-        data_in = Dataset("./test/transform_hub_in")
-        ds_out = ds
-        ds_out.create_tensor("image")
-        ds_out.create_tensor("label")
-        transform(data_in, [fn2], ds_out)
-        data_in.delete()
-        assert len(ds_out) == 100
-        for index in range(100):
-            np.testing.assert_array_equal(
-                ds_out[index].image.numpy(), index * np.ones((100, 100))
-            )
-            np.testing.assert_array_equal(
-                ds_out[index].label.numpy(), index * np.ones((1,))
-            )
-
-            
-def check_transform_on_ds(data_in: Dataset, ds_out: Dataset):
+    with Dataset("./test/transform_hub_in_generic") as data_in:
```

Comment:
What happened to the isolated filesystem? Could you address the tests that use local paths? It's really annoying not to have it in the `tmp` directory.

Note also that this will run even without the `--local` flag

Reply:
No good way to prevent this from running right now, we can revamp the fixtures in the next sprint to allow 2 datasets as input, I think there's already a task for it.